SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -1824,3 +1827,103 @@ void USpatialNetDriver::DelayedSendDeleteEntityRequest(Worker_EntityId EntityId,
 		Sender->SendDeleteEntityRequest(EntityId);
 	}, Delay, false);
 }
+
+void USpatialNetDriver::HandleStartupOpQueueing(const TArray<Worker_OpList*>& InOpLists)
```

Comment:
I'm trying to discern what happens in the non authoritative server case. Is it the case that non auth servers are queuing until the auth server has set the BeginPlay flag on the GSM?

Reply:
There is an assumption in this change that for the initial GSM update from SpatialOS -> worker, the following will happen:

 - For auth server, it will get both the BeginPlay flag and the GSM Auth in the same crit section
 - For non-auth server, it will get only BeginPlay flag

In this case, we can call BeginPlay on the non-auth server even if the BeginPlay() flag is false.  We just need to have received the flag (and not the auth-update) to realize we are on a non-auth server.  That's what it will do for non-auth case.

Reply:
This means of course non-auth servers can call BeginPlay() before auth server does, but I didn't see a problem with this happening.  Let me know if the assumption is incorrect :)

Reply:
The only purpose now of the begin play flag (which is now named AuthBeginPlayCalled) is to let the restarting auth server know that it should call BeginPlay() without authority

Reply:
> For auth server, it will get both the BeginPlay flag Op and the GSM SAMC Auth Op in the same crit section

I know for certain I don't think we can rely on this. It is very possible, that the authority op isn't in the critical section. It sometimes will be, but also sometimes won't.

Reply:
Ok.  I thought we had this guarantee for the initial update, but not for subsequent updates.  Let's chat tomorrow.

Reply:
Also, I think it will be minor changes to the PR if it's the case

Reply:
From reading this thread on worker-support, I believe we will receive the 2 ops in the same OpList, which is actually the only requirement:
https://improbable.slack.com/archives/CE5BYA3HD/p1561985578164900

Reply:
Sahil and I have discussed this further and we believe this will work correctly.  The uncertainty was introduced by me mentioning the assumption that for the auth server, the 2 necessary Ops would be received in the same critical section, which is not guaranteed.  The only actual requirement is that they are received in the same oplist, which should be the case according to David A. here:
https://improbable.slack.com/archives/CE5BYA3HD/p1562528935199300